### ISSUE TITLE
Blank client screen after a match finishes

### ISSUE THREAD

**Issue by kyechou** [kyechou/leagueoflegends#92]
### System information

 - Distro: Arch Linux
 - DE/WM/Compositor: bspwm
 - CPU: Intel i7-9750H
 - GPU: NVIDIA GeForce RTX 2060
 - Graphics card drivers installed: `nvidia-open-dkms`, `nvidia-utils`, `lib32-nvidia-utils`

<details>
<summary>vulkaninfo --summary</summary>
<br>

```
==========
VULKANINFO
==========

Vulkan Instance Version: 1.3.246


Instance Extensions: count = 21
-------------------------------
VK_EXT_acquire_drm_display             : extension revision 1
VK_EXT_acquire_xlib_display            : extension revision 1
VK_EXT_debug_report                    : extension revision 10
VK_EXT_debug_utils                     : extension revision 2
VK_EXT_direct_mode_display             : extension revision 1
VK_EXT_display_surface_counter         : extension revision 1
VK_KHR_device_group_creation           : extension revision 1
VK_KHR_display                         : extension revision 23
VK_KHR_external_fence_capabilities     : extension revision 1
VK_KHR_external_memory_capabilities    : extension revision 1
VK_KHR_external_semaphore_capabilities : extension revision 1
VK_KHR_get_display_properties2         : extension revision 1
VK_KHR_get_physical_device_properties2 : extension revision 2
VK_KHR_get_surface_capabilities2       : extension revision 1
VK_KHR_portability_enumeration         : extension revision 1
VK_KHR_surface                         : extension revision 25
VK_KHR_surface_protected_capabilities  : extension revision 1
VK_KHR_wayland_surface                 : extension revision 6
VK_KHR_xcb_surface                     : extension revision 6
VK_KHR_xlib_surface                    : extension revision 6
VK_LUNARG_direct_driver_loading        : extension revision 1

Instance Layers: count = 5
--------------------------
VK_LAYER_NV_optimus               NVIDIA Optimus layer         1.3.236  version 1
VK_LAYER_VALVE_steam_fossilize_32 Steam Pipeline Caching Layer 1.3.207  version 1
VK_LAYER_VALVE_steam_fossilize_64 Steam Pipeline Caching Layer 1.3.207  version 1
VK_LAYER_VALVE_steam_overlay_32   Steam Overlay Layer          1.3.207  version 1
VK_LAYER_VALVE_steam_overlay_64   Steam Overlay Layer          1.3.207  version 1

Devices:
========
GPU0:
	apiVersion         = 1.3.236
	driverVersion      = 530.41.3.192
	vendorID           = 0x10de
	deviceID           = 0x1f11
	deviceType         = PHYSICAL_DEVICE_TYPE_DISCRETE_GPU
	deviceName         = NVIDIA GeForce RTX 2060
	driverID           = DRIVER_ID_NVIDIA_PROPRIETARY
	driverName         = NVIDIA
	driverInfo         = 530.41.03
	conformanceVersion = 1.3.3.1
	deviceUUID         = 807b6ae2-4c4f-0da1-d45f-e5144767918d
	driverUUID         = 3f4f90dd-753c-577a-b215-5e16a1ae82c0
```

</details>

### Describe the issue

After a match finishes, the client's screen shows up completely dark/blank. Sometimes it recovers after a few seconds. Sometimes it just hangs.

### Steps to reproduce

1. Install lol-git with wine-lol 8.5
2. Install the game as described.
3. Run `leagueoflegends start` to start a game.
4. When a game finishes, the issue occurs as described above.

(TODO: Attach the verbose log output)

[edit: 2023-04-11 10:58 CT]

People from discord:
> The client crashing after game is something to do with the honor a teammate screen. If you level up after a game and let the honor select page time out the client won't crash.

**Comment by HerrmannMax** [comment]
Greetings,
i have the same Issue.

### System information
+ Distro: Nobara Linux
+ CPU: AMD Ryzen 5 2600X
+ GPU: NVIDIA GeForce GTX 1070 TI
+ Graphics card drivers installed: nvidia-driver.x86_64=3:530.41.03-1.fc37, nvidia-gpu-firmware.noarch=20230404-149.fc37

<details>
<summary>
vulkaninfo --summary
</summary>

```bash
==========
VULKANINFO
==========

Vulkan Instance Version: 1.3.240


Instance Extensions: count = 22
-------------------------------
VK_EXT_acquire_drm_display             : extension revision 1
VK_EXT_acquire_xlib_display            : extension revision 1
VK_EXT_debug_report                    : extension revision 10
VK_EXT_debug_utils                     : extension revision 2
VK_EXT_direct_mode_display             : extension revision 1
VK_EXT_display_surface_counter         : extension revision 1
VK_EXT_surface_maintenance1            : extension revision 1
VK_EXT_swapchain_colorspace            : extension revision 4
VK_KHR_device_group_creation           : extension revision 1
VK_KHR_display                         : extension revision 23
VK_KHR_external_fence_capabilities     : extension revision 1
VK_KHR_external_memory_capabilities    : extension revision 1
VK_KHR_external_semaphore_capabilities : extension revision 1
VK_KHR_get_display_properties2         : extension revision 1
VK_KHR_get_physical_device_properties2 : extension revision 2
VK_KHR_get_surface_capabilities2       : extension revision 1
VK_KHR_portability_enumeration         : extension revision 1
VK_KHR_surface                         : extension revision 25
VK_KHR_surface_protected_capabilities  : extension revision 1
VK_KHR_wayland_surface                 : extension revision 6
VK_KHR_xcb_surface                     : extension revision 6
VK_KHR_xlib_surface                    : extension revision 6

Instance Layers: count = 10
---------------------------
VK_LAYER_FROG_gamescope_wsi       Gamescope WSI (XWayland Bypass) Layer (x86_64) 1.3.221  version 1
VK_LAYER_MANGOHUD_overlay         Vulkan Hud Overlay                             1.3.0    version 1
VK_LAYER_MESA_device_select       Linux device selection layer                   1.3.211  version 1
VK_LAYER_MESA_overlay             Mesa Overlay layer                             1.3.211  version 1
VK_LAYER_NV_optimus               NVIDIA Optimus layer                           1.3.236  version 1
VK_LAYER_VALVE_steam_fossilize_32 Steam Pipeline Caching Layer                   1.3.207  version 1
VK_LAYER_VALVE_steam_fossilize_64 Steam Pipeline Caching Layer                   1.3.207  version 1
VK_LAYER_VALVE_steam_overlay_32   Steam Overlay Layer                            1.3.207  version 1
VK_LAYER_VALVE_steam_overlay_64   Steam Overlay Layer                            1.3.207  version 1
VK_LAYER_VKBASALT_post_processing a post processing layer                        1.3.223  version 1

Devices:
========
GPU0:
	apiVersion         = 1.3.236
	driverVersion      = 530.41.3.192
	vendorID           = 0x10de
	deviceID           = 0x1b82
	deviceType         = PHYSICAL_DEVICE_TYPE_DISCRETE_GPU
	deviceName         = NVIDIA GeForce GTX 1070 Ti
	driverID           = DRIVER_ID_NVIDIA_PROPRIETARY
	driverName         = NVIDIA
	driverInfo         = 530.41.03
	conformanceVersion = 1.3.3.1
	deviceUUID         = 78cd434a-1526-0719-fb87-9e76bd587005
	driverUUID         = 3f4f90dd-753c-577a-b215-5e16a1ae82c0
GPU1:
	apiVersion         = 1.3.246
	driverVersion      = 0.0.1
	vendorID           = 0x10005
	deviceID           = 0x0000
	deviceType         = PHYSICAL_DEVICE_TYPE_CPU
	deviceName         = llvmpipe (LLVM 15.0.7, 256 bits)
	driverID           = DRIVER_ID_MESA_LLVMPIPE
	driverName         = llvmpipe
	driverInfo         = Mesa 23.1.0-devel (LLVM 15.0.7)
	conformanceVersion = 1.3.1.1
	deviceUUID         = 6d657361-3233-2e31-2e30-2d6465766500
	driverUUID         = 6c6c766d-7069-7065-5555-494400000000
```
</details>

Here is also the log from Lutris:
[Lol-wine-frozen-client.log](https://github.com/kyechou/leagueoflegends/files/11201980/Lol-wine-frozen-client.log)

I started the client, entered a custom game, left and then the client froze.
I think at line 589 is the point, where I left the game.

Thanks for the great work, may can help you!

**Comment by kyechou** [comment]
The issue seems to be a regression somewhere between 7.0 and 7.14 (https://github.com/GloriousEggroll/wine-ge-custom/releases/tag/7.0-GE-5-LoL). At the moment users should wait for 7.0 releases (https://github.com/GloriousEggroll/wine-ge-custom/pull/11).

**Comment by kyechou** [comment]
For people who want to play the game, this issue is resolved in [the GE 7.0-8 LoL release](https://github.com/GloriousEggroll/wine-ge-custom/releases/tag/7.0-GE-8-LoL). A PR is opened for review at https://github.com/M-Reimer/wine-lol/pull/117, so the `wine-lol` and `wine-lol-bin` AUR packages should also be updated soon.

The issue will be left open for fixing the problem in more recent versions of wine.

**Comment by jhundley9109** [comment]
Hello,

I've been testing building wine myself and I have been testing with an almost identical build you have here with 8.5. I have some notes:

I suspect 04-proton-LAA_staging.patch isn't necessary anymore since it's a 64bit application?
lfh-non-proton-pre-needed.patch and 50-proton_LFH.patch certainly shouldn't be necessary anymore and can be removed permanently since wine 8.3. See https://www.winehq.org/announce/8.3

I have noticed with my build and I will test the this build (https://github.com/GloriousEggroll/wine/tree/ge-lol-8.5.1-testing) but I suspect it to be the same, CPU usage is higher on Wine 8 compared to Wine 7 during game. I don't have hard/fast measurements for this but my laptop CPU fan suggests to me this is the case LOL.

I have noticed wine 8 does not support fsync or I don't know how to get it to work with it. Honestly I don't know the details of fsync vs esync but I have compiled wine multiple times attempting different patches including manually applying patches myself. See https://github.com/jhundley9109/wine/commits/wine-8.5-lol-testing-fix-fsync

It may be possible my higher cpu usage I'm noticing on Wine 8 is related to fsync not enabled on Wine 8? Just some thoughts on it at least.

Great work getting the patches out and the work you put in the last week for the league community!
